portability-types-common/src/main/java/org/datatransferproject/types/common/models/music/MusicContainerResource.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2022 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.datatransferproject.types.common.models.music;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import org.datatransferproject.types.common.models.ContainerResource;
+
+/**
+ * A Wrapper for a group of music entities.
+ */
+@JsonTypeName("MusicContainerResource")
+public class MusicContainerResource extends ContainerResource {
+
+  public enum MusicPlatform {
```

Comment:
this is a red flag; this is very much crossing boundaries between adapters and models. a models entire purpose is to be agnostic to the platform that leverages it.

particularly having it as an enum makes it seem like one can rely on this. If this is just going to be used as pass-through information to be displayed to the user, might I suggest putting a URL field here instead? eg:

```java
/**
 * Optionally indicates the platform from which this playlist originated, if appropriate, as
 * added context in any affordance displayed directly to end-users of a transfer.
 *
 * For example, fictional `Acme Music` adapter authors (a product `https://music.acme.co`) might
 * choose to encode `acme.co` here, `music.acme.co/playlist/3231` or may even choose to
 * leave this blank.
 */
Some.URL.Type uriOriginPlatform;
```

Happy to chat more about this - sorry I didn't catch this before in any discussions.

Reply:
We found out that we would like to have originalPlatform information for almost all the music related objects (MusicPlaylist, MusicPlaylistItem, MusicRecording and MusicRelease) to keep track of importing operations. Another option is to keep the originalPlatform in each music object, which might be redundant considering MusicPlaylistItem(MusicRecording(MusicRelease)) will have the same information.

Reply:
Removed the enum in the latest commit https://github.com/google/data-transfer-project/pull/1131/commits/62fc7fa1d4432560f2dbe2133228287d6c3b471c.